### DISCUSSION TITLE
venus-market design & roadmap

### DISCUSSION THREAD

**Discussion by Joss-Hua** [filecoin-project/venus#4532] [Announcements]
~~## venus-market roadmap~~
~~Current long term vision for the development of venus-market~~

~~## Objectives~~
~~1. venus-market as independent module (short term): allow venus storage provider to accept and retrieve deals~~
~~2. venus-market as shared module: let venus-market be the gateway for storage providers to accept and retrieve deals~~

~~## Milestones~~
~~Interoperability with Lotus in accepting deals: allow lotus client to send deal to venus-market~~
~~Interoperability with Lotus in retrieval: allow lotus client to retrieve deal from venus-market~~
~~Compact API: venus-market local store~~
~~Data transfer support for different protocols in addition to `Graphsync`~~
~~Deal meta data stored locally by venus-market~~
~~venus-market as deal gateway for storage providers using venus chain services~~
~~Deal match making: multiple copies for store and retrieval~~
~~venus-market as gateway for IPFS: options for paid IPFS node~~
~~Lightweight client: compatibility with Lotus and support for venus-market unique features~~

***

### This is for discussing the [venus-market module design & roadmap](https://github.com/filecoin-project/venus/blob/master/documentation/venus-market%20module%20design%20%26%20roadmap.md)

As the rebranding of filecoin terminology spearheaded by FIP0018 settled, consensus has been reached across communities (developers, providers, ecosystem partners and etc) to push for taking on more storage deals to improve the public perception on the fact that most of the network storages are still commited capacities (CCs). Given the above sentiment, design and implemenation of venus-market module has been put into the spot light. A clear long-term roadmap is due for Venus community to dissus and iterate on, also as a means for better communications with filecoin eocsystem in general.

While Lotus is leading the way of implementing a full-fledged market module according to the spec, Venus has been making much efforts to catch up and closing the gap in regard to markets. Right now, Lotus supports a 1-to-1 (client-to-provider) storage and retrieval model where burdens of discoveries are mostly on storage clients and match making services like Estuary. Negotiation is a fairly mannual process and does not support much flexibility. As venus-team is picking up the reminiscences of the Filecoin Component Architecture, emergent ways of how market could facilitate the dynamics between storage providers and storage clients are constatntly being intergrated into the long-term vision of Venus filecoin.

Current roadmap for [venus-market](https://github.com/filecoin-project/venus/blob/master/documentation/venus-market%20module%20design%20%26%20roadmap.md) are loosely broken into the 3 phases.
- Phase 1: peer-to-peer model (short-term)
- Phase 2: platform model
- Phase 3: peer-to-peer model (long-term vision)

If you have any questions wrt the roadmap, leave a comment in the [discussion thread](https://github.com/filecoin-project/venus/discussions/4711)!

**Comment by Fatman13** [comment]
Updating market phase 2 roadmap...

![image](https://user-images.githubusercontent.com/1591330/149718213-aea7df14-fb28-4b46-88d8-7e1c40f796e5.png)
